compiler/src/dmd/opover.d

```diff
@@ -594,6 +594,9 @@ Expression opOverloadAssign(AssignExp e, Scope* sc)
 
 Expression opOverloadBinary(BinExp e, Scope* sc)
 {
+    if (Expression err = binSemanticProp(e, sc))
+        return err;
```

Comment:
Why call the variable `err`? does `binSemanticProp` return either an `ErrorExp` or `null`?

Reply:
Yes, that's also what confused me in the original code. I wondered why semantic would seemingly return after binSemanticProp, but the function doesn't actually return the result of succesfull semantic on the BinExp's left/right child nodes. Instead, it modifies them, and returns null on success. It only returns a non-null `ErrorExp()` on error, and then you're supposed to return early. I hoped calling the variable `err` would make that clearer (I didn't want to use the same name as the very common `error` function).